### PR TITLE
Nigel/feature/iteration4

### PR DESCRIPTION
**Pull request by Pocketzs**
Passed and created top revenue earners method for iteration 4 using TDD.  All helper methods are tested and spec harness passes.

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -168,4 +168,30 @@ def total_revenue_by_date(date)
...
+  end
+
+  def ranked_merchants 
+    ranked_merchants_with_revenue.map do |merch_arr|
+      merch_arr[0]
+    end
+  end
+
+  def upper_bound(x) 
+    x-1
```
Comment (beddings81) [inline lib/sales_analyst.rb]: add spaces ex: x - 1

File: spec/merchant_spec.rb [hunk 2]
```diff
@@ -25,4 +26,35 @@
...
+      :merchants => './data/merchants.csv',
+      :invoices  => './data/invoices.csv',
+      :invoice_items => './data/invoice_items.csv',
+      :transactions => './data/transactions.csv',
+      :customers => './data/customers.csv'
+      )
+
+      m = sales_engine.merchants.all[333]
+
+      expect(m.total_revenue).to eq(0.1263009e6)
```
Comment (KaraJoHo) [inline spec/merchant_spec.rb]: Indentation check

File: lib/sales_analyst.rb [hunk 3]
```diff
@@ -168,4 +168,30 @@ def total_revenue_by_date(date)
...
+  end
+
+  def top_revenue_earners(x = 20)
```
Comment (Pocketzs) [inline lib/sales_analyst.rb]: The interaction patten uses x?  Should we change it?

Comment (KaraJoHo) [reply lib/sales_analyst.rb]: I vote yes

**Review by beddings81 — APPROVED** [review]
Looks good!

**Review by Kledin85 — APPROVED** [review]
looks good

**Comment by Pocketzs** [comment]
Needed hound comments have been addressed

**Review by KaraJoHo — APPROVED** [review]
I like the helper methods!

**Pull request merged**